src/main/java/Duke.java

```diff
@@ -1,10 +1,223 @@
+    }
+
+    /* Validate whether input is a number */
+    private static boolean checkIsNumber(String input) {
+        String regex = "[0-9]+";
+        Pattern pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
+        Matcher matcher = pattern.matcher(input);
+
+        return matcher.find();
+    }
+
+    /* Show mark task as done message */
+    private static void showMarkAsDoneMessage(Task task) {
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println("  " + task);
+    }
+
+    /* Show invalid input format error */
+    private static void showInvalidInputFormatError() {
+        System.out.println("Invalid input format. Please try again with correct format.");
+    }
+
+    /* Show invalid index out of bound error */
+    private static void showIndexOutOfBoundError() {
+        System.out.println("Invalid index number. Please enter the correct index number.");
+    }
+
+    /* Show invalid command error */
+    private static void showInvalidCommandError() {
+        System.out.println("Invalid command. Please try again with valid command.");
+    }
+
+    /* Parse input into command command */
+    private static void parseCommand(String input) {
+        int firstSpacePosition = input.indexOf(" ");
+        String lowercaseCommand = input;
+        String parameter = "";
+
+        if (firstSpacePosition > 0) {
+            lowercaseCommand = input.substring(0, firstSpacePosition).toLowerCase();
+            parameter = input.substring(firstSpacePosition + 1);
+        }
+
+        executeCommand(lowercaseCommand, parameter);
+    }
+
+    /* Execute command based on input from parseCommand */
+    private static void executeCommand(String command, String parameter) {
+        switch (command) {
+        case "list":
+            showTasks();
+            break;
+        case "done":
+            markAsDone(parameter);
+            break;
+        case "todo":
+            createTodoTask(parameter);
+            break;
+        case "event":
+            createEventTask(parameter);
+            break;
+        case "deadline":
+            createDeadlineTask(parameter);
+            break;
+        default:
+            showInvalidCommandError();
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        showGreeting();
+
+        Scanner line = new Scanner(System.in);
+
+        while (line.hasNextLine()) {
+            String input = line.nextLine();
+            showDivider();
+            if (input.toLowerCase().equals("bye")) {
+                break;
+            }
+            parseCommand(input);
+            showDivider();
+        }
```

Comment:
Should this be extracted out?

Reply:
Thank you for pointing this out! I will extract it out in the next commits.